tests/unit/obuf-limits.tcl

```diff
@@ -70,4 +70,32 @@ start_server {tags {"obuf-limits"}} {
         assert {$omem >= 100000 && $time_elapsed < 6}
         $rd1 close
     }
+
+    test {Won't write replies if client output buffer hard limit is enforced} {
+        r config set client-output-buffer-limit {normal 100000 0 0}
+
+        # Total size of all items must be more than 2 * 100000
+        for {set i 0} {$i < 100000} {incr i} {
+            r lpush mylist $i $i
+        }
+        set orig_mem [s used_memory]
+
+        # Get all items
+        set rd [redis_deferring_client]
+        $rd lrange mylist 0 -1
```

Comment:
we may need `$rd flush` and even a short sleep if we wanna be sure redis processed this before the next statement.
and it would also be good idea to add a check that makes sure redis disconnected the client before we proceed to do `$rd read`. we can do that by looking at `[s connected_clients]` (comparing before and other), or name it and look for it in `CLIENT LIST`.